stubs/html5lib/html5lib/serializer.pyi

```diff
@@ -32,11 +33,11 @@ class HTMLSerializer:
     errors: Incomplete
     strict: bool
     def __init__(self, **kwargs) -> None: ...
-    def encode(self, string): ...
-    def encodeStrict(self, string): ...
+    def encode(self, string: text_type) -> str: ...
+    def encodeStrict(self, string: text_type) -> str: ...
     encoding: Incomplete
-    def serialize(self, treewalker, encoding=None) -> Generator[Incomplete]: ...
-    def render(self, treewalker, encoding=None): ...
+    def serialize(self, treewalker: str, encoding: str) -> str: ...
```

Comment:
1. Indeed, `encoding` could accept `str`, but it is an optional argument (this is what causes problems with CI passing).
2. `treewalker` looks like some `Iterable` of dictionaries (i'd call it `_Token`), but since we don't have `_Token` TypedDict yet, I suggest using `Iterable[dict[str, Incomplete]]`
3. `serialize` doesn't just return a string, it yields it, and since with the encoding set, it can be bytes. I think you can use `Iterator` or `Generator`
```suggestion
    def serialize(self, treewalker: Iterable[dict[str, Incomplete]], encoding: str | None = None) -> Iterator[str | bytes]: ...
```